pybamm/expression_tree/independent_variable.py

```diff
@@ -118,8 +126,13 @@ class SpatialVariable(IndependentVariable):
     """
 
     def __init__(
-        self, name, domain=None, auxiliary_domains=None, domains=None, coord_sys=None
-    ):
+        self,
+        name: str,
+        domain: list[str] | str | None = None,
```

Comment:
The types for the different domains (domain, auxilary_domains and domains) are also repeated a lot. As with `ChildValue`/`ChildSymbol` these should be defined in on place and used throughout.